### PR TITLE
Remove airnode address from template verification

### PR DESCRIPTION
**Pull request by vponline**
This removes airnode address as an input for templateId derivation.

### CODE REVIEW

**Review by Siegrift — APPROVED** [review]
👍 LGTM.

File: packages/airnode-node/src/evm/templates/template-verification.ts [hunk 1]
```diff
@@ -8,7 +8,6 @@ interface ValidatedField {
 
 function getTemplateIdValidationFields(template: ApiCallTemplateWithoutId): ValidatedField[] {
   return [
-    { type: 'address', value: template.airnodeAddress },
```
Comment (amarthadan) [inline packages/airnode-node/src/evm/templates/template-verification.ts:11]: I think @acenolaza [is correct](https://api3dao.atlassian.net/browse/AN-703?focusedCommentId=15321). The `getExpectedTemplateId` function is used in `api/index.ts` (https://github.com/api3dao/airnode/blob/master/packages/airnode-node/src/api/index.ts#L149) both for signed data gateway requests but also regular Airnode RRP requests. That's why you had to change tests everywhere, not just the ones for `callApi`. I think this change breaks the Airnode RRP. We need two different template ID derivation functions, one for Airnode RRP (v0 contracts) and one for signed data gateway (v1 contracts).

**Review by amarthadan — CHANGES_REQUESTED** [review]

File: packages/airnode-node/src/evm/templates/template-verification.ts (hunk 1 above)
Comment (vponline) [reply packages/airnode-node/src/evm/templates/template-verification.ts:11]: Do you prefer to have two separate functions, or we could pass the request `type` (regular/http-signed-gateway) into `getExpectedTemplateId` and return based on that?

Comment (amarthadan) [reply packages/airnode-node/src/evm/templates/template-verification.ts:11]: I think two separate functions will result in a nicer code.

**Review by amarthadan — APPROVED** [review]
LGTM :+1:

**Pull request merged**